tilewe/engine.py

```diff
@@ -80,9 +80,9 @@ def _play_game(self, player_to_engine: list[int]) -> tuple[list[int], list[int],
                 board.push(move) 
 
             # put scores back in original engine order 
-            winners = [ engine_to_player[x] for x in board.winners ]
+            winners = [ player_to_engine[x] for x in board.winners ]
             board_scores = board.scores
-            scores = [ board_scores[player_to_engine[i]] for i in range(len(self.engines)) ]
+            scores = [ board_scores[engine_to_player[i]] for i in range(len(self.engines)) ]
```

Comment:
`board.scores` is the board's perspective of players not engines, so when indexing it from the engine order perspective we need to map from engines to players